perf/src/packet.rs

```diff
@@ -153,6 +153,7 @@ impl BytesPacket {
 pub enum PacketBatch {
     Pinned(PinnedPacketBatch),
     Bytes(BytesPacketBatch),
+    WithClientId(BytesPacketBatchWithClientId),
```

Comment:
@vadorovsky for me to have more context, wasn't this structure `PacketBatch` introduced to help migration from Pinned to Bytes? So the idea to eclipse Pinned variant at some moment in time.